src/components/PrintModal/WatermarkModal/WatermarkModal.js

```diff
@@ -0,0 +1,434 @@
+import ColorPalette from 'components/ColorPalette';
+import Slider from 'components/Slider';
+import { circleRadius } from 'constants/slider';
+import core from 'core';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withTranslation } from 'react-i18next';
+import './WatermarkModal.scss';
+
+import ActionButton from 'components/ActionButton';
+
+const DESIRED_WIDTH = 300;
+const DESIRED_HEIGHT = 300;
+
+// numbers were taken from font dropdown menu in google docs
+const FONT_SIZES = [
+  8, 9, 10, 11, 12, 14, 16, 18, 24, 30, 36, 48,
+];
+
+const WATERMARK_LOCATIONS = {
+  CENTER: 'Center',
+  TOP_LEFT: 'Top Left',
+  TOP_RIGHT: 'Top Right',
+  TOP_CENTER: 'Top Center',
+  BOT_LEFT: 'Bottom Left',
+  BOT_RIGHT: 'Bottom Right',
+  BOT_CENTER: 'Bottom Center',
+};
+
+const FORM_FIELD_KEYS = {
+  location: 'location',
+  fontSize: 'fontSize',
+  text: 'text',
+  color: 'color',
+  opacity: 'opacity',
+};
+
+const DEFAULT_VALS = {
+  [FORM_FIELD_KEYS.location]: WATERMARK_LOCATIONS.CENTER,
+  [FORM_FIELD_KEYS.fontSize]: FONT_SIZES[FONT_SIZES.length / 2],
+  [FORM_FIELD_KEYS.text]: '',
+  [FORM_FIELD_KEYS.color]: new window.Annotations.Color(241, 160, 153),
+  [FORM_FIELD_KEYS.opacity]: 100,
+};
+
+class WatermarkModal extends React.PureComponent {
+  handleWatermarkRenderFxn;
+
+  static propTypes = {
+    isVisible: PropTypes.bool,
+    pageIndexToView: PropTypes.number,
+    modalClosed: PropTypes.func,
+    formSubmitted: PropTypes.func,
+    t: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    const locationSettings = this.initializeLocationSettings();
+    this.state = {
+      isVisible: false,
```

Comment:
I don't really think we need this state because `this.state.isVisible` isn't used in the render function or anywhere else. I think we can just use `this.props.isVisible` so that this component is fully controlled by the `PrintModal`? What do you think?

Reply:
good catch.
i agree with you about removing "isVIsible" from the state because its not being used.
Since it is not being used at all, it might be confusing to other devs if we kept it there.

Reply:
i have removed it.